src/server/multi_command_squasher.cc

```diff
@@ -254,19 +264,23 @@ bool MultiCommandSquasher::ExecuteSquashed(facade::RedisReplyBuilder* rb) {
   uint64_t after_hop = proactor->GetMonotonicTimeNs();
   bool aborted = false;
 
+  size_t size = 0;
   for (auto idx : order_) {
     auto& replies = sharded_[idx].replies;
     CHECK(!replies.empty());
 
     aborted |= opts_.error_abort && CapturingReplyBuilder::TryExtractError(replies.back());
 
-    current_reply_size_.fetch_sub(Size(replies.back()), std::memory_order_relaxed);
```

Comment:
I said nothing when `current_reply_size_` was added. it was a mistake. I do not want anyone introduces global states in Dragonfly codebase.

Reply:
@BorysTheDev  FYI

Reply:
but @romange  we discussed this when current_reply_size_ was added. Because the multi command sqasher is adding replies in different threads you said it makes sense.

Reply:
Would something like this work?
https://github.com/dragonflydb/dragonfly/compare/RemoveAtomic?expand=1

Reply:
@romange The change in your branch linked above is that you count the reply size after we executed all the squashed commands. So it does uses the thread local approach correctly but what we expose to metrics is not accurate because at the time the capture reply builders grow we do not expose this until we finish with all the squashed. I think that applying such logic will impact the throttling on reply size just in delay, we will throttle but not when actually we are at the threshold but with some delay. I guess we can do this change.

Reply:
There is no a "correct" solution because I can show you a scenario where single central atomic won't work either:
we throttle before we send commands to shards, but maybe there are tons of squashed commands in flight that have not filled their replies yet, so you let the next command pass and only then the reply buffer increases.

I would rather have a less accurate metric than have have all our threads contend on atomics and now on a single condvar. this kills performance. I won't be surprised that even now squashing performance is worse because of the "reply bufffer size" atomic being hammered by multiple threads.